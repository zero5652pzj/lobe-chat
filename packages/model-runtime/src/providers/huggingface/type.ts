interface HFArchitecture {
  input_modalities: string[];
  output_modalities: string[];
}

interface HFProviderInfo {
  context_length?: number;
  is_model_author?: boolean;
  pricing?: {
    input?: number;
    output?: number;
  };
  provider: string;
  status: string;
  supports_structured_output?: boolean;
  supports_tools?: boolean;
}

export interface HuggingFaceRouterModelCard {
  architecture: HFArchitecture;
  created: number;
  id: string;
  object: string;
  owned_by: string;
  providers: HFProviderInfo[];
}

export interface HuggingFaceRouterResponse {
  data: HuggingFaceRouterModelCard[];
  object: string;
}
