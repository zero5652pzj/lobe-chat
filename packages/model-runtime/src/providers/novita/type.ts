export interface NovitaModelCard {
  context_size: number;
  created: number;
  description: string;
  features?: string[];
  id: string;
  input_modalities?: string[];
  input_token_price_per_m: number;
  max_output_tokens?: number;
  model_type?: string;
  output_token_price_per_m: number;
  status: number;
  tags: string[];
  title: string;
}
