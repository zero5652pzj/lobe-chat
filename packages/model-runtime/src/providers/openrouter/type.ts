interface ModelPricing {
  completion: string;
  image?: string;
  input_cache_read?: string;
  input_cache_write?: string;
  prompt: string;
  request?: string;
  web_search?: string;
  internal_reasoning?: string;
}

interface TopProvider {
  context_length: number;
  max_completion_tokens: number | null;
  is_moderated: boolean;
}

interface Architecture {
  modality: string;
  input_modalities: string[];
  output_modalities: string[];
  tokenizer: string;
  instruct_type: string | null;
}

export interface OpenRouterModelCard {
  id: string;
  canonical_slug: string;
  hugging_face_id?: string;
  name: string;
  created: number;
  description?: string;
  context_length: number;
  architecture: Architecture;
  pricing: ModelPricing;
  top_provider: TopProvider;
  per_request_limits?: any | null;
  supported_parameters: string[];
  default_parameters?: any | null;
}

interface OpenRouterOpenAIReasoning {
  effort: 'high' | 'medium' | 'low';
  exclude?: boolean;
}

interface OpenRouterAnthropicReasoning {
  exclude?: boolean;
  max_tokens: number;
}

interface OpenRouterCommonReasoning {
  exclude?: boolean;
}

export type OpenRouterReasoning =
  | OpenRouterOpenAIReasoning
  | OpenRouterAnthropicReasoning
  | OpenRouterCommonReasoning;
