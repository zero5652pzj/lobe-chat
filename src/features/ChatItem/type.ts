import { AlertProps, AvatarProps, DivProps, MarkdownProps } from '@lobehub/ui';
import { EditableMessageProps, MetaData } from '@lobehub/ui/chat';
import { ReactNode } from 'react';
import { FlexboxProps } from 'react-layout-kit';

export interface ChatItemProps extends Omit<FlexboxProps, 'children' | 'onChange'> {
  aboveMessage?: ReactNode;
  actions?: ReactNode;
  actionsWrapWidth?: number;
  avatar: MetaData;
  avatarAddon?: ReactNode;
  avatarProps?: AvatarProps;
  belowMessage?: ReactNode;
  /**
   * @description Whether the chat item is disabled
   * @default false
   */
  disabled?: boolean;
  /**
   * @description Whether the chat item is in editing mode
   */
  editing?: boolean;
  /**
   * @description Props for Error render
   */
  error?: AlertProps;
  errorMessage?: ReactNode;
  fontSize?: number;
  /**
   * @description Whether the chat item is in loading state
   */
  loading?: boolean;
  markdownProps?: Omit<MarkdownProps, 'className' | 'style' | 'children'>;
  /**
   * @description The message content of the chat item
   */
  message?: ReactNode;
  messageExtra?: ReactNode;
  onAvatarClick?: () => void;
  onDoubleClick?: DivProps['onDoubleClick'];
  /**
   * @default "..."
   */
  placeholderMessage?: string;
  /**
   * @description The placement of the chat item
   * @default 'left'
   */
  placement?: 'left' | 'right';
  /**
   * @description Whether the chat item is primary
   */
  primary?: boolean;
  renderMessage?: (content: ReactNode) => ReactNode;
  /**
   * @description Whether to hide the avatar
   * @default false
   */
  showAvatar?: boolean;
  /**
   * @description Whether to show the title of the chat item
   */
  showTitle?: boolean;
  text?: EditableMessageProps['text'];
  /**
   * @description The timestamp of the chat item
   */
  time?: number;
  titleAddon?: ReactNode;
  /**
   * @description The type of the chat item
   * @default 'bubble'
   */
  variant?: 'bubble' | 'docs';
}
