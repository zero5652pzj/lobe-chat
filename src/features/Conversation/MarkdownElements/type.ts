import { FC, ReactNode } from 'react';

export interface MarkdownElementProps<T = any> {
  children: ReactNode;
  id: string;
  node: {
    properties: T;
  };
  tagName: string;
  type: string;
}

export type MarkdownPluginScope = 'user' | 'assistant' | 'all';

export interface MarkdownElement {
  Component: FC<MarkdownElementProps>;
  rehypePlugin?: any;
  remarkPlugin?: any;
  scope: MarkdownPluginScope;
  tag: string;
}
