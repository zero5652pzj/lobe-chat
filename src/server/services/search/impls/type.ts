import { SearchParams, UniformSearchResponse } from '@lobechat/types';

/**
 * Search service implementation interface
 */
export interface SearchServiceImpl {
  /**
   * Query for search results
   */
  query(query: string, params?: SearchParams): Promise<UniformSearchResponse>;
}
