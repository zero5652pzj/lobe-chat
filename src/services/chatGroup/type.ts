import {
  ChatGroupAgentItem,
  ChatGroupItem,
  NewChatGroup,
  NewChatGroupAgent,
} from '@/database/schemas';

export interface IChatGroupService {
  addAgentsToGroup(groupId: string, agentIds: string[]): Promise<ChatGroupAgentItem[]>;
  createGroup(params: Omit<NewChatGroup, 'userId'>): Promise<ChatGroupItem>;
  deleteGroup(id: string): Promise<any>;
  getGroup(id: string): Promise<ChatGroupItem | undefined>;
  getGroupAgents(groupId: string): Promise<ChatGroupAgentItem[]>;
  getGroups(): Promise<ChatGroupItem[]>;
  removeAgentsFromGroup(groupId: string, agentIds: string[]): Promise<any>;
  updateAgentInGroup(
    groupId: string,
    agentId: string,
    updates: Partial<Pick<NewChatGroupAgent, 'order' | 'role'>>,
  ): Promise<ChatGroupAgentItem>;
  updateGroup(id: string, value: Partial<ChatGroupItem>): Promise<ChatGroupItem>;
}
