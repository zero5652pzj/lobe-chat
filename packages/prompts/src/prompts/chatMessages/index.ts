import { UIChatMessage } from '@lobechat/types';

const chatMessage = (message: UIChatMessage) => {
  return `<${message.role}>${message.content}</${message.role}>`;
};

export const chatHistoryPrompts = (messages: UIChatMessage[]) => {
  return `<chat_history>
${messages.map((m) => chatMessage(m)).join('\n')}
</chat_history>`;
};

export const groupSupervisorPrompts = (messages: UIChatMessage[]) => {
  const formatMessage = (message: UIChatMessage) => {
    const author = message.role === 'user' ? 'user' : message.agentId || 'assistant';
    const targetAttr = message.targetId ? ` target="${message.targetId}"` : '';
    return `<message author="${author}"${targetAttr}>${message.content}</message>`;
  };

  // Filter out supervisor's own messages to avoid self-referencing
  const filteredMessages = messages.filter((message) => message.role !== 'supervisor');

  return filteredMessages.map((m) => formatMessage(m)).join('\n');
};

export const groupMemeberSpeakingPrompts = (messages: UIChatMessage[]) => {
  return `<chat_group>
${messages.map((m) => chatMessage(m)).join('\n')}
</chat_group>`;
};

/**
 * Filters messages that an agent should see based on DM targeting rules
 * - Agent sees all group messages (no targetId)
 * - Agent sees DMs where they are the target
 * - Agent sees DMs they sent
 * - Agent sees user messages that are group messages or targeted to them
 * - For DM messages not involving the agent, content is replaced with "***"
 *
 * TODO: Use context engineering to filter messages
 */
export const filterMessagesForAgent = (
  messages: UIChatMessage[],
  agentId: string,
): UIChatMessage[] => {
  return messages
    .filter((message) => {
      // Exclude supervisor messages (messages with role="supervisor")
      if (message.role === 'supervisor') {
        return false;
      }
      return true;
    })
    .map((message) => {
      // Always include system messages as-is
      if (message.role === 'system') {
        return message;
      }

      // For user messages, check DM targeting rules
      if (message.role === 'user') {
        // If no target specified, it's a group message - include it as-is
        if (!message.targetId) {
          return message;
        }

        // If the message is targeted to this agent, include it as-is
        if (message.targetId === agentId) {
          return message;
        }

        // Otherwise, it's a DM to another agent - replace content with "***"
        return {
          ...message,
          content: '***',
        };
      }

      // For assistant messages, check DM targeting rules
      if (message.role === 'assistant') {
        // If no target specified, it's a group message - include it as-is
        if (!message.targetId) {
          return message;
        }

        // If the agent is the target of the DM, include it as-is
        if (message.targetId === agentId) {
          return message;
        }

        // If the agent sent the message, include it as-is
        if (message.agentId === agentId) {
          return message;
        }

        // Otherwise, it's a DM not involving this agent - replace content with "***"
        return {
          ...message,
          content: '***',
        };
      }

      // Default: include the message as-is
      return message;
    });
};

/**
 * Consolidates group chat message history into a single formatted string
 * for use in system messages. Each message is formatted as "(AuthorName): content"
 */
export const consolidateGroupChatHistory = (
  messages: UIChatMessage[],
  agents: { id: string; title: string }[] = [],
) => {
  if (messages.length === 0) return '';

  // Create a map for quick agent lookup
  const agentMap = new Map(agents.map((agent) => [agent.id, agent.title]));

  const formatMessage = (message: UIChatMessage) => {
    let authorName: string;

    if (message.role === 'user') {
      authorName = 'User';
    } else if (message.role === 'assistant' && message.agentId) {
      // Try to get agent title from the provided agents map
      authorName = agentMap.get(message.agentId) || `Agent ${message.agentId}`;
    } else {
      authorName = 'Assistant';
    }

    return `(${authorName}): ${message.content}`;
  };

  return messages
    .filter((m) => m.content && m.content.trim()) // Filter out empty messages
    .map(formatMessage)
    .join('\n');
};
