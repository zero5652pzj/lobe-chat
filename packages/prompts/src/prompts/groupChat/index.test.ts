import type { UIChatMessage } from '@lobechat/types';
import { beforeEach, describe, expect, it, vi } from 'vitest';

import { filterMessagesForAgent } from '../chatMessages';
import { buildGroupChatSystemPrompt } from './index';

describe('buildGroupChatSystemPrompt', () => {
  const baseSystemRole = 'You are an expert collaborator.';
  const mockTimestamp = 1634567890000;

  beforeEach(() => {
    vi.spyOn(Date, 'now').mockReturnValue(mockTimestamp);
  });

  const messages: UIChatMessage[] = [
    {
      id: 'm1',
      role: 'user',
      content: 'Hello everyone',
      createdAt: mockTimestamp,
      updatedAt: mockTimestamp,
      meta: {},
    },
    {
      id: 'm2',
      role: 'assistant',
      agentId: 'agent-1',
      content: 'Hi! I can help with that.',
      createdAt: mockTimestamp,
      updatedAt: mockTimestamp,
      meta: {},
    },
  ];

  it('should generate correct prompt with members and history', () => {
    const result = buildGroupChatSystemPrompt({
      agentId: 'agent-1',
      baseSystemRole,
      groupMembers: [
        { id: 'user', title: 'User Name' },
        { id: 'agent-1', title: 'Agent One' },
      ],
      messages,
    });

    expect(result).toMatchSnapshot();
  });

  it('should generate correct prompt without members and history', () => {
    const result = buildGroupChatSystemPrompt({
      agentId: 'agent-2',
      baseSystemRole,
      groupMembers: [],
      messages: [],
    });

    expect(result).toMatchSnapshot();
  });

  it('should include response instruction for group message', () => {
    const result = buildGroupChatSystemPrompt({
      agentId: 'agent-1',
      baseSystemRole,
      groupMembers: [{ id: 'agent-1', title: 'Agent One' }],
      messages,
    });

    expect(result).toContain("Now it's your turn to respond");
    expect(result).toContain('the group publicly');
  });

  it('should include response instruction for direct message', () => {
    const result = buildGroupChatSystemPrompt({
      agentId: 'agent-1',
      baseSystemRole,
      groupMembers: [{ id: 'agent-1', title: 'Agent One' }],
      messages,
      targetId: 'user',
    });

    expect(result).toContain("Now it's your turn to respond");
    expect(result).toContain('user');
  });

  it('should include supervisor instruction when provided', () => {
    const result = buildGroupChatSystemPrompt({
      agentId: 'agent-1',
      baseSystemRole,
      groupMembers: [{ id: 'agent-1', title: 'Agent One' }],
      messages,
      instruction: 'Please be concise',
    });

    expect(result).toContain('SUPERVISOR INSTRUCTION: Please be concise');
  });
});

describe('filterMessagesForAgent', () => {
  const mockTimestamp = 1634567890000;
  const agentId = 'agent-1';

  beforeEach(() => {
    vi.spyOn(Date, 'now').mockReturnValue(mockTimestamp);
  });

  const createMessage = (
    id: string,
    role: 'user' | 'assistant' | 'system',
    content: string,
    options?: { agentId?: string; targetId?: string },
  ): UIChatMessage => ({
    id,
    role,
    content,
    createdAt: mockTimestamp,
    updatedAt: mockTimestamp,
    meta: {},
    ...options,
  });

  it('should handle system messages', () => {
    const messages = [createMessage('1', 'system', 'System message')];
    const result = filterMessagesForAgent(messages, agentId);
    expect(result).toMatchSnapshot();
  });

  it('should handle group messages without targetId', () => {
    const messages = [
      createMessage('1', 'user', 'Hello everyone'),
      createMessage('2', 'assistant', 'Hi there', { agentId: 'agent-2' }),
    ];
    const result = filterMessagesForAgent(messages, agentId);
    expect(result).toMatchSnapshot();
  });

  it('should handle user direct messages', () => {
    const messages = [
      createMessage('1', 'user', 'Private message to agent-1', { targetId: agentId }),
      createMessage('2', 'user', 'Private message to agent-2', { targetId: 'agent-2' }),
    ];
    const result = filterMessagesForAgent(messages, agentId);
    expect(result).toMatchSnapshot();
  });

  it('should handle assistant direct messages', () => {
    const messages = [
      createMessage('1', 'assistant', 'DM to agent-1', { agentId: 'agent-2', targetId: agentId }),
      createMessage('2', 'assistant', 'DM to agent-2', { agentId: 'agent-3', targetId: 'agent-2' }),
    ];
    const result = filterMessagesForAgent(messages, agentId);
    expect(result).toMatchSnapshot();
  });

  it('should handle messages sent by this agent', () => {
    const messages = [
      createMessage('1', 'assistant', 'My message to agent-2', { agentId, targetId: 'agent-2' }),
      createMessage('2', 'assistant', 'My group message', { agentId }),
      createMessage('3', 'assistant', 'Other agent message', {
        agentId: 'agent-2',
        targetId: 'agent-3',
      }),
    ];
    const result = filterMessagesForAgent(messages, agentId);
    expect(result).toMatchSnapshot();
  });

  it('should replace content for DMs not involving this agent', () => {
    const messages = [
      createMessage('1', 'user', 'Secret message', { targetId: 'agent-2' }),
      createMessage('2', 'assistant', 'Private response', { agentId: 'agent-2', targetId: 'user' }),
      createMessage('3', 'assistant', 'Agent to agent DM', {
        agentId: 'agent-2',
        targetId: 'agent-3',
      }),
    ];
    const result = filterMessagesForAgent(messages, agentId);
    expect(result).toMatchSnapshot();
  });

  it('should handle mixed message types', () => {
    const messages = [
      createMessage('1', 'system', 'System prompt'),
      createMessage('2', 'user', 'Hello everyone'),
      createMessage('3', 'assistant', 'Hi!', { agentId }),
      createMessage('4', 'user', 'DM to agent-1', { targetId: agentId }),
      createMessage('5', 'assistant', 'DM response', { agentId, targetId: 'user' }),
      createMessage('6', 'user', 'Secret to agent-2', { targetId: 'agent-2' }),
      createMessage('7', 'assistant', 'Secret response', { agentId: 'agent-2', targetId: 'user' }),
    ];
    const result = filterMessagesForAgent(messages, agentId);
    expect(result).toMatchSnapshot();
  });

  it('should preserve all message properties when replacing content', () => {
    const originalMessage = createMessage('1', 'user', 'Secret message', { targetId: 'agent-2' });
    originalMessage.files = ['file1'];
    originalMessage.tools = [];
    originalMessage.parentId = 'parent1';
    originalMessage.observationId = 'obs1';

    const result = filterMessagesForAgent([originalMessage], agentId);
    expect(result).toMatchSnapshot();
  });
});
