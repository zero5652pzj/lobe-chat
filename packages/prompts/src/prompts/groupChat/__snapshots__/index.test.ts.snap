// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildGroupChatSystemPrompt > should generate correct prompt with members and history 1`] = `
"You are an expert collaborator.

Guidelines:

- Stay in character as agent-1 (Agent One)
- Be concise and natural, behave like a real person
- The group supervisor will decide whether to send it privately or publicly, so you just need to say the actuall content, even it's a DM to a specific member. Do not pretend you've sent it.
- Be collaborative and build upon others' responses when appropriate
- Keep your responses concise and relevant to the ongoing discussion

<group_members>
[
  {
    "id": "user",
    "title": "User Name"
  },
  {
    "id": "agent-1",
    "title": "Agent One"
  }
]
</group_members>

Now it's your turn to respond.  You are sending message to the group publicly. Please respond as this agent would, considering the full conversation history provided above. Directly return the message content, no other text. You do not need add author name or anything else."
`;

exports[`buildGroupChatSystemPrompt > should generate correct prompt without members and history 1`] = `
"You are an expert collaborator.

Guidelines:

- Stay in character as agent-2 (Agent)
- Be concise and natural, behave like a real person
- The group supervisor will decide whether to send it privately or publicly, so you just need to say the actuall content, even it's a DM to a specific member. Do not pretend you've sent it.
- Be collaborative and build upon others' responses when appropriate
- Keep your responses concise and relevant to the ongoing discussion

Now it's your turn to respond.  You are sending message to the group publicly. Please respond as this agent would, considering the full conversation history provided above. Directly return the message content, no other text. You do not need add author name or anything else."
`;

exports[`filterMessagesForAgent > should handle assistant direct messages 1`] = `
[
  {
    "agentId": "agent-2",
    "content": "DM to agent-1",
    "createdAt": 1634567890000,
    "id": "1",
    "meta": {},
    "role": "assistant",
    "targetId": "agent-1",
    "updatedAt": 1634567890000,
  },
  {
    "agentId": "agent-3",
    "content": "***",
    "createdAt": 1634567890000,
    "id": "2",
    "meta": {},
    "role": "assistant",
    "targetId": "agent-2",
    "updatedAt": 1634567890000,
  },
]
`;

exports[`filterMessagesForAgent > should handle group messages without targetId 1`] = `
[
  {
    "content": "Hello everyone",
    "createdAt": 1634567890000,
    "id": "1",
    "meta": {},
    "role": "user",
    "updatedAt": 1634567890000,
  },
  {
    "agentId": "agent-2",
    "content": "Hi there",
    "createdAt": 1634567890000,
    "id": "2",
    "meta": {},
    "role": "assistant",
    "updatedAt": 1634567890000,
  },
]
`;

exports[`filterMessagesForAgent > should handle messages sent by this agent 1`] = `
[
  {
    "agentId": "agent-1",
    "content": "My message to agent-2",
    "createdAt": 1634567890000,
    "id": "1",
    "meta": {},
    "role": "assistant",
    "targetId": "agent-2",
    "updatedAt": 1634567890000,
  },
  {
    "agentId": "agent-1",
    "content": "My group message",
    "createdAt": 1634567890000,
    "id": "2",
    "meta": {},
    "role": "assistant",
    "updatedAt": 1634567890000,
  },
  {
    "agentId": "agent-2",
    "content": "***",
    "createdAt": 1634567890000,
    "id": "3",
    "meta": {},
    "role": "assistant",
    "targetId": "agent-3",
    "updatedAt": 1634567890000,
  },
]
`;

exports[`filterMessagesForAgent > should handle mixed message types 1`] = `
[
  {
    "content": "System prompt",
    "createdAt": 1634567890000,
    "id": "1",
    "meta": {},
    "role": "system",
    "updatedAt": 1634567890000,
  },
  {
    "content": "Hello everyone",
    "createdAt": 1634567890000,
    "id": "2",
    "meta": {},
    "role": "user",
    "updatedAt": 1634567890000,
  },
  {
    "agentId": "agent-1",
    "content": "Hi!",
    "createdAt": 1634567890000,
    "id": "3",
    "meta": {},
    "role": "assistant",
    "updatedAt": 1634567890000,
  },
  {
    "content": "DM to agent-1",
    "createdAt": 1634567890000,
    "id": "4",
    "meta": {},
    "role": "user",
    "targetId": "agent-1",
    "updatedAt": 1634567890000,
  },
  {
    "agentId": "agent-1",
    "content": "DM response",
    "createdAt": 1634567890000,
    "id": "5",
    "meta": {},
    "role": "assistant",
    "targetId": "user",
    "updatedAt": 1634567890000,
  },
  {
    "content": "***",
    "createdAt": 1634567890000,
    "id": "6",
    "meta": {},
    "role": "user",
    "targetId": "agent-2",
    "updatedAt": 1634567890000,
  },
  {
    "agentId": "agent-2",
    "content": "***",
    "createdAt": 1634567890000,
    "id": "7",
    "meta": {},
    "role": "assistant",
    "targetId": "user",
    "updatedAt": 1634567890000,
  },
]
`;

exports[`filterMessagesForAgent > should handle system messages 1`] = `
[
  {
    "content": "System message",
    "createdAt": 1634567890000,
    "id": "1",
    "meta": {},
    "role": "system",
    "updatedAt": 1634567890000,
  },
]
`;

exports[`filterMessagesForAgent > should handle user direct messages 1`] = `
[
  {
    "content": "Private message to agent-1",
    "createdAt": 1634567890000,
    "id": "1",
    "meta": {},
    "role": "user",
    "targetId": "agent-1",
    "updatedAt": 1634567890000,
  },
  {
    "content": "***",
    "createdAt": 1634567890000,
    "id": "2",
    "meta": {},
    "role": "user",
    "targetId": "agent-2",
    "updatedAt": 1634567890000,
  },
]
`;

exports[`filterMessagesForAgent > should preserve all message properties when replacing content 1`] = `
[
  {
    "content": "***",
    "createdAt": 1634567890000,
    "files": [
      "file1",
    ],
    "id": "1",
    "meta": {},
    "observationId": "obs1",
    "parentId": "parent1",
    "role": "user",
    "targetId": "agent-2",
    "tools": [],
    "updatedAt": 1634567890000,
  },
]
`;

exports[`filterMessagesForAgent > should replace content for DMs not involving this agent 1`] = `
[
  {
    "content": "***",
    "createdAt": 1634567890000,
    "id": "1",
    "meta": {},
    "role": "user",
    "targetId": "agent-2",
    "updatedAt": 1634567890000,
  },
  {
    "agentId": "agent-2",
    "content": "***",
    "createdAt": 1634567890000,
    "id": "2",
    "meta": {},
    "role": "assistant",
    "targetId": "user",
    "updatedAt": 1634567890000,
  },
  {
    "agentId": "agent-2",
    "content": "***",
    "createdAt": 1634567890000,
    "id": "3",
    "meta": {},
    "role": "assistant",
    "targetId": "agent-3",
    "updatedAt": 1634567890000,
  },
]
`;
